### ISSUE TITLE
Augmenting testing set for multi-turn chitchat tasks

### ISSUE THREAD

**Issue by zhongpeixiang** [facebookresearch/ParlAI#2349]
Given a multi-turn conversation in a test set: x<sub>1</sub>, x<sub>2</sub>, ..., x<sub>N</sub>, where x<sub>N</sub> is the final response. Can we create N-1 test conversations, similar to the training data augmentation in (Lowe, et. al. 2016)? namely 

x<sub>1</sub>, x<sub>2</sub>, 

x<sub>1</sub>, x<sub>2</sub>, x<sub>3</sub>, 

...

x<sub>1</sub>, x<sub>2</sub>, ..., x<sub>N</sub>

In this way, we can evaluate our models more thoroughly since there are more testing examples. 

Any drawbacks for this evaluation?

```
Lowe, Ryan, et al. "The Ubuntu Dialogue Corpus: A Large Dataset for Research in Unstructured Multi-Turn Dialogue Systems." Proceedings of the 16th Annual Meeting of the Special Interest Group on Discourse and Dialogue. 2015.
```

**Comment by stephenroller** [comment]
Hi Peixiang,

Unless I'm misunderstanding you, this is generally how we evaluate almost all datasets in ParlAI already. Each full length conversation is called as an "episode", and each individual turn is called an "example". So as you see `num_examples` in your training logs, this is the number of *turns* the model has seen.

For some datasets, we even have $N$ test examples, instead of $N-1$, as we insert a special `__SILENCE__` token to indicate that the model must produce the first response.

Some datasets only have $N/2$ examples though. These are roughly equivalent to the model only learning to speak $x_2, x_4, x_6$, ... PersonaChat and Wizard of Wikipedia like this. Others have both sides of the conversation (that is, one time the model plays speaker A, another time it plays speaker B). ConvAI2 and DailyDialog are both like this.

Does this answer your question? Please feel free to reopen if it I have misunderstood you're wishing for something different.
